core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -162,14 +162,25 @@ private int getUsableCapacity() {
    */
   public void free() {
     if (consumer != null) {
-      consumer.freeArray(array);
+      if (null != array) {
+        consumer.freeArray(array);
+      }
       array = null;
     }
   }
 
   public void reset() {
     if (consumer != null) {
       consumer.freeArray(array);
+      // this is needed to prevent a 'nested' spill,
```

Comment:
nit: it doesn't prevent a nested spill, it only renders it harmless
remove this line - the rest of the comment is true.